api/Dockerfile

```diff
@@ -0,0 +1,27 @@
+FROM php:7.4.27-apache-bullseye as base
+
+# Install pdo_pgsql extension.
+RUN apt-get update \
+    && apt-get install -y libpq-dev \
+    && docker-php-ext-install pdo_pgsql
+
+WORKDIR /var/www/html/api
+
+COPY .env.example .env
```

Comment:
If .env already exists, wouldn't we want to use that instead?

Reply:
```
COPY .env.example .env
# ...
COPY . .
```

This pattern ensures that (1) `.env` is first put in place from `.env.example`, but (2) if `.env` also exists, it overrides it when the rest of the files are copied over. (assuming `.env` is in `.gitignore`, but not `.dockerignore`)

The downside of this: Since it works with or without .env, it's easy to forget when you have a custom `.env` present. You might think you're creating a fresh build, but your lingering custom `.env` is still building something different than if you'd `git clone`d a fresh repo.

This is kinda a consistent gotcha with gitignoring any important config, but complicated if developer doesn't know where docker is getting it :) (there's an argument not to do it, because being overly clever is a bad idea)

Maybe this is more legible and hints at what's happening...?

```
COPY .env.example .env
# ...
COPY . *.env .
```

It's the same as `COPY . .`, but explicitly points to `.env` in a way that won't fail like `COPY . .env .` would:

Reply:
Ah, good point, `COPY . .` will work.
But in that case, why manually copy .env.example so much earlier in the dockerfile?

Reply:
Good call! Fixed! 

Sometimes things will be in odd places in dockerfiles (which might not be as logically ordered as normal scripts) to make best use of caching, but this wasn't the case here :)